### PR TITLE
Justin/filtering

### LINKED ISSUE #33: [US 04.02.01] Filter mood events by emotion
**Description:** As a participant, I want to filter my mood history list to show only mood events with a particular emotional state.

**Rationale:** So that I can easily find mood events by emotions, and understand situations that triggered a certain emotion.

### PR DESCRIPTION
**Pull request by toado**
Closes #33

**Summary** 
Users can now filter mood event by specified emotional states

### CODE REVIEW

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FilterActivity.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+                if (selectedFilters.size() > 0) {
+                    returnIntent.putExtra("filters", selectedFilters);
+                }
+                setResult(Activity.RESULT_OK, returnIntent);
+                finish();
+            }
+        });
+    }
+
+    /** Initialize the contents of the Activity's standard options menu.
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FilterActivity.java]: This line should be on a new line

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FollowRequestActivity.java [hunk 2]
```diff
@@ -49,11 +49,11 @@ public void onClick(View view) {
         });
 
         service = new UserService();
-        service.getFollowRequests(this);
+//        service.getFollowRequests(this);
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FollowRequestActivity.java]: Is this required for later functionality? If not, it'd be cleaner to remove it

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IMoodEventServiceProvider.java [hunk 3]
```diff
@@ -92,15 +92,15 @@
      * method with the new mood history list when a change occurs.
      * @param listener The listener to pass the new mood history list to
      */
-    void getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener);
+    ListenerRegistration getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener);
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IMoodEventServiceProvider.java:96]: Javadoc for the the return needed (`@return`)

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IMoodEventServiceProvider.java [hunk 4]
```diff
@@ -92,15 +92,15 @@
...
 
     /**
      * Listen to mood history updates of the current user. Calls the listener's onMoodHistoryUpdate
      * method with the new mood history list when a change occurs.
      * @param listener The listener to pass the new mood history list to
-     * @param filterBy The {@link EmotionalState} to filter the list by.
+     * @param filterBy The list of {@link EmotionalState} to filter the list by.
      */
-    void getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener, EmotionalState filterBy);
+    ListenerRegistration getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener, ArrayList<String> filterBy);
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IMoodEventServiceProvider.java:105]: `@return` missing

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/MoodEventService.java [hunk 5]
```diff
@@ -207,73 +209,51 @@ public void onFailure(@NonNull Exception e) {
...
-
     /**
      * Listen to mood history updates of the current user. Calls the listener's onMoodHistoryUpdate
      * method with the new mood history list when a change occurs.
+     *
      * @param listener The listener to pass the new mood history list to
      */
     @Override
-    public void getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener) {
+    public ListenerRegistration getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener) {
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/MoodEventService.java:250]: Missing `@return` in Javadoc

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/MoodEventService.java [hunk 6]
```diff
@@ -207,73 +209,51 @@ public void onFailure(@NonNull Exception e) {
...
      * Listen to mood history updates of the current user. Calls the listener's onMoodHistoryUpdate
      * method with the new mood history list when a change occurs.
+     *
      * @param listener The listener to pass the new mood history list to
-     * @param filterBy The {@link EmotionalState} to filter the list by.
+     * @param filterBy The list of {@link EmotionalState} to filter the list by.
      */
     @Override
-    public void getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener, EmotionalState filterBy) {
+    public ListenerRegistration getMoodHistoryUpdates(OnMoodHistoryUpdateListener listener, ArrayList<String> filterBy) {
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/MoodEventService.java:269]: `@return`

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/MoodEventService.java [hunk 7]
```diff
@@ -207,73 +209,51 @@ public void onFailure(@NonNull Exception e) {
...
     }
 
     /**
      * Run the specified Mood History query and notify the listener on completion
      * @param moodHistoryQuery The query to run
      * @param listener The listener to pass the new mood history list to
      */
-    private void runMoodHistoryQuery(Query moodHistoryQuery, final OnMoodHistoryUpdateListener listener) {
-        moodHistoryQuery.addSnapshotListener(new EventListener<QuerySnapshot>() {
+    private ListenerRegistration runMoodHistoryQuery(Query moodHistoryQuery, final OnMoodHistoryUpdateListener listener) {
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/MoodEventService.java:287]: `@return`

**Review by WJXHenry — CHANGES_REQUESTED** [review]
Mostly (I think all of them are) Javadoc stuff

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FollowRequestActivity.java (hunk 2 above)
Comment (toado) [reply Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FollowRequestActivity.java]: me didnt work on this part :octopus:

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FilterActivity.java [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+                }
+                setResult(Activity.RESULT_OK, returnIntent);
+                finish();
+            }
+        });
+    }
+
+    /**
+     * Initialize the contents of the Activity's standard options menu.
+     * @param menu The options menu in which you place your items.
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/activities/FilterActivity.java:216]: Sorry I missed one more `@return`

**Review by WJXHenry — CHANGES_REQUESTED** [review]
One more!

**Review by WJXHenry — DISMISSED** [review]
LGTM

**Review by WJXHenry — APPROVED** [review]
LGTM

**Pull request merged**